staging/src/k8s.io/kubectl/pkg/cmd/top/top_node.go

```diff
@@ -43,42 +42,13 @@ type TopNodeOptions struct {
 	SortBy          string
 	NoHeaders       bool
 	NodeClient      corev1client.CoreV1Interface
-	HeapsterOptions HeapsterTopOptions
-	Client          *metricsutil.HeapsterMetricsClient
 	Printer         *metricsutil.TopCmdPrinter
 	DiscoveryClient discovery.DiscoveryInterface
 	MetricsClient   metricsclientset.Interface
 
 	genericclioptions.IOStreams
 }
 
-type HeapsterTopOptions struct {
-	Namespace string
-	Service   string
-	Scheme    string
-	Port      string
-}
-
-func (o *HeapsterTopOptions) Bind(flags *pflag.FlagSet) {
-	if len(o.Namespace) == 0 {
-		o.Namespace = metricsutil.DefaultHeapsterNamespace
-	}
-	if len(o.Service) == 0 {
-		o.Service = metricsutil.DefaultHeapsterService
-	}
-	if len(o.Scheme) == 0 {
-		o.Scheme = metricsutil.DefaultHeapsterScheme
-	}
-	if len(o.Port) == 0 {
-		o.Port = metricsutil.DefaultHeapsterPort
-	}
-
-	flags.StringVar(&o.Namespace, "heapster-namespace", o.Namespace, "Namespace Heapster service is located in")
```

Comment:
I do understand your approach and even though these flags might not be used, we are still obliged to follow the [deprecation policies](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). Please then deprecate these flags and we'll be able to remove then a year from now.

Reply:
Ok, would it make sense to discuss exception in this case? Those flags are used exclusively component that was deprecated 6 and retired 4 releases ago. We should try to answer question "Do we have a chance to break users?". In my opinion no.

I'm for removing those flags as I had multiple users being confused by kubectl still mentioning Heapster. All kubernetes documentation already discourages usage of Heapster.

Reply:
Sorry, no exception, we need to be clear, even if that means holding these flags even when they are not doing anything. Users need the time to remove them.

Reply:
Ok, will prepare separate PR focused on deprecation.

Reply:
Ok, I updated this PR to leave heapster flags as no-op.